src/nanoarrow/nanoarrow_device.c

```diff
@@ -175,10 +175,12 @@ ArrowErrorCode ArrowDeviceArrayInitAsync(struct ArrowDevice* device,
     return device->array_init(device, device_array, array, sync_event, stream);
   }
 
-  // Handling a sync event or stream is not supported in the default constructor
-  if (sync_event != NULL || stream != NULL) {
-    return EINVAL;
-  }
+  // TODO: Right now we have to let these events through and ignore them
+  // because we don't have a good way to separate the source and destination
+  // streams.
+  // if (sync_event != NULL || stream != NULL) {
+  //   return EINVAL;
+  // }
```

Comment:
The purpose of using a sync event is so that we don't need to care about the source and destination streams. We just need to tell the destination stream to wait for the event (if it exists) before passing it to the memcpy calls. If there's no sync_event, then we don't need to be concerned about the source stream at all

Reply:
That sounds like a good solution and handling to me, though as far as taking the stream argument is concerned. The stream should always be considered to be "the stream that any operations should be put on", i.e. the destination. By using the whole sync-event handling that we do, we bypass the need to logically separate a source stream and destination stream: either we need to sync (there should be a sync event and the stream should be given a `cuStreamWaitEvent` on it) or there is no sync-event and thus no need to sync. The stream is always the stream to be used for any memcpy operations. Just mentioning for future reference to hopefully clear this up and make it easier to understand. 

obviously @kkraus14 can correct me if i'm wrong here... lol